dwa_local_planner/src/dwa_planner.cpp

```diff
@@ -465,8 +470,9 @@ namespace dwa_local_planner {
     // Other storage
     double distance_from_robot = -1;
     double minimum_distance = 0.5; // it needs to be at least this close
-
-    for (size_t k = 0; k < global_plan_.size() - 1; ++k)
+    double plan_skipped = 0; //local plan skipped
```

Comment:
I'm trying to figure out what is going on here.  I think there is an issue with plan distance on larger maps.

Reply:
So the function originally calculated the distance of the local plan by first finding the robots shortest distance to the path and then adding up all the segment length after that. This is needed so that the path_distance isnt only updated every 2hz. Since the global plan is not stored with the same local plan tf, I found the 'skipped distance' using the local path.I then added the entire global path and subtracted this skipped distance.

Steps on what this does:

1)  Calculate local path skipped and shortest distance to local path.
2)  Calculate entire global path distance.
3) Subtract to find actual distance to goal.